generator/src/main/java/com/scottlogic/deg/generator/fieldspecs/FieldSpec.java

```diff
@@ -200,8 +194,35 @@ public FieldSpec withMustContainRestriction(MustContainRestriction mustContainRe
             this.source);
     }
 
-    public FieldSpecSource getFieldSpecSource() {
-        return this.source;
+    /** Create a predicate that returns TRUE for all (and only) values permitted by this FieldSpec */
+    public boolean permits(@NotNull Object value) {
```

Comment:
My preference would be to remove this from `FieldSpec` and place it on a new type; this can then be passed by reference instead of the `Predicate` in `SetRestrictionsMergeOperation`.